docs/specification.md

```diff
@@ -0,0 +1,1199 @@
+<!--
+SPDX-FileCopyrightText: 2020 tqtezos
+SPDX-License-Identifier: MIT
+-->
+
+**Overview**
+------------
+
+These specifications were assembled with the following references:
+
+- [*CENTRE Fiat Token design*](https://github.com/centrehq/centre-tokens/blob/78d964a1a8d481ffd8152772d7a66e47df54b3db/doc/tokendesign.md) which is served as a reference specification to tezos stablecoin smart contract.
+
+- Tezos Token Standard:
+  [*FA2*](https://gitlab.com/tzip/tzip/-/blob/76d5f3791bfbfe3c9bf95ad5ec5fc6cbeeca2d0e/proposals/tzip-12/tzip-12.md) and
+  [*ManagedLedger*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-7/ManagedLedger.md)
+
+- [*Michelson Contract Interfaces and Conventions*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-4/tzip-4.md) TZIP which defines `view` and `void` type synonyms
+
+**General Requirements**
+------------------------
+
+- The token(s) must be and FA2 compatible as to facilitate listing on
+  exchanges and interaction with services which support FA2
+
+**Ledger**
+==========
+
+Every address that is stored in ledger is associated with its
+current balance, token id and current minting allowance.
```

Comment:
> token id

The contract should hold only one type of tokens. Storing `token id` for each address does not make sense, it is always 0. Some other places should be updated as well.